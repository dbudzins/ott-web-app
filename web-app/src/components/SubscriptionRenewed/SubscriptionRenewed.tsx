import React from 'react';
import { useTranslation } from 'react-i18next';

import Button from '../Button/Button';
import type { Subscription } from '../../../types/subscription';
import { formatDate, formatPrice } from '../../utils/formatting';
import type { Customer } from '../../../types/account';

import styles from './SubscriptionRenewed.module.scss';

type Props = {
  customer: Customer;
  subscription: Subscription;
  onClose: () => void;
};

const SubscriptionRenewed: React.FC<Props> = ({ onClose, customer, subscription }: Props) => {
  const { t } = useTranslation('account');
  const date = formatDate(subscription.expiresAt);
  const price = formatPrice(subscription.nextPaymentPrice, subscription.nextPaymentCurrency, customer.country);

  return (
    <div>
      <h2 className={styles.title}>{t('subscription_renewed.title')}</h2>
      <p className={styles.paragraph}>{t('subscription_renewed.message', { date, price })}</p>
      <Button label={t('subscription_renewed.back_to_profile')} onClick={onClose} fullWidth />
    </div>
  );
};

export default SubscriptionRenewed;
