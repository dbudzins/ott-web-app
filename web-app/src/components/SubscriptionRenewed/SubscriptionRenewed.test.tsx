import React from 'react';
import { render } from '@testing-library/react';

import customer from '../../fixtures/customer.json';
import subscription from '../../fixtures/subscription.json';
import type { Subscription } from '../../../types/subscription';
import type { Customer } from '../../../types/account';

import SubscriptionRenewed from './SubscriptionRenewed';

describe('<SubscriptionRenewed>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<SubscriptionRenewed customer={customer as Customer} subscription={subscription as Subscription} onClose={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
