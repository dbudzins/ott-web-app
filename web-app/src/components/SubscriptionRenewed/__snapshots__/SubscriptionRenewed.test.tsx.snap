// Vitest Snapshot v1

exports[`<SubscriptionRenewed> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="title"
    >
      subscription_renewed.title
    </h2>
    <p>
      subscription_renewed.message
    </p>
    <button
      class="button default outlined fullWidth"
    >
      <span>
        subscription_renewed.back_to_profile
      </span>
    </button>
  </div>
</div>
`;
